src/App.tsx

```diff
@@ -1,18 +1,37 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState(1);
+  const [perPage, setPerPage] = useState(5);
+  const optionNumbers = [3, 5, 10, 20];
+
+  const firstItem = (currentPage - 1) * perPage + 1;
+  const lastItem = Math.min(currentPage * perPage, 42);
+
+  const visibleElements = () => {
+    const firstPageIndex = (currentPage - 1) * perPage;
+    const lastPageIndex = firstPageIndex + perPage;
+
+    return items.slice(firstPageIndex, lastPageIndex);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page
+        {currentPage}
+        (items
+        {` ${firstItem} - ${lastItem} `}
+        of 42)
```

Comment:
Consider using [Template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for better readability